docs/contracts.rst

```diff
@@ -428,8 +428,20 @@ change by overriding).
 Constant State Variables
 ************************
 
-State variables can be declared as constant (this is not yet implemented
-for array and struct types and not possible for mapping types).
+State variables can be declared as constant. In this case, they have to be
+assigned a value or expression which is a constant at compile time. Expressions
+that might have a side-effect on memory allocation are allowed, but those that
+might have a side-effect on other memory objects are not. This makes it possible
```

Comment:
Side-effect (read/write) on storage is not allowed.  It's allowed to consume gas.  It's not allowed to look up the remaining gas.